operational_analysis/methods/plant_analysis.py

```diff
@@ -28,6 +28,36 @@
 
 logger = logging.getLogger(__name__)
 
+
+def get_annual_values(data):
+    """
+    This function returns annual summations of values in a pandas Series (or each column of a pandas DataFrame) with a
+    DatetimeIndex index starting from the first row. The purpose of the function is to correctly resample to annual
+    values when the first index does not fall on the beginning of the month.
+
+    Args:
+        data(:obj:`pandas.Series or pandas.DataFrame`): Input data with a DatetimeIndex index.
+
+    Returns:
+        :obj:`numpy.ndarray`: Array containing annual summations for each column of the input data.
+    """
+    
+    # shift time index to beginning of first month so resampling by 'MS' groups the data into full years 
+    # starting from the beginning
+    data.index = (
+        data.index - pd.Timedelta(
+            data.index[0]
+            - (
+                data.index[0].floor('d')
+                + pd.offsets.MonthEnd(0)
+                - pd.offsets.MonthBegin(1)
+            )
+        )
+    )
```

Comment:
Because it's in a function I might just rewrite it as the following just for the sake of clarity in what is going on in the code:
```python
    ix_start = data.index[0]
    month_start = ix_start.floor("d") + pd.offsets.MonthEnd(0) - pd.offsets.MonthBegin(1)
    data.index = data.index - pd.Timedelta(ix_start - month_start)
```

Reply:
Thanks Rob, I made both of those changes you suggested.